File: 백준/Silver/20365. 블로그2/블로그2.py
import sys
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
def colorMatch(color, n):
    
    count =1
    drawing = [False for _ in range (n)]
    
    for i in range(n):
        if colors[i]==color:
            drawing[i]=True
    idx = -1
    for i in range(n):
        if drawing[i]==False:
            if idx<0:
                idx = i
                count+=1
            elif idx==(i-1):
                idx = i
                continue
            else:
                idx = i
                count+=1
    return count

n = int(input())
colors = list(map(str, input().strip('\n')))
b_count = colorMatch('B', n)
r_count= colorMatch('R', n)
print(min(b_count, r_count))

File: 백준/Silver/20365. 블로그2/블로그2.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func findStr(c string, n int, str []string) (count int) {
	count = 1
	var drawing = make([]bool, n)
	var index = -1
	for i, v := range str {
		if c == v {
			drawing[i] = true
		}
	}
	for i, _ := range str {
		if drawing[i] == false {
			if index < 0 {
				count += 1
				index = i
			} else if index == i-1 {
				index = i
			} else {
				index = i
				count += 1
			}
		}
	}
	return count
}
func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()
	var n int
	var str string
	fmt.Fscanf(reader, "%d\n%s", &n, &str)
	strArr := strings.Split(str, "")
	RCount := findStr("R", n, strArr)
	BCount := findStr("B", n, strArr)
	if RCount < BCount {
		fmt.Fprintf(writer, "%d\n", RCount)
	} else {
		fmt.Fprintf(writer, "%d\n", BCount)
	}

}
